### PR TITLE
[OpenCL SYCL] Add support for triSYCL in TensorFlow

### PR DESCRIPTION
**Pull request by a-doumoulakis**
Add the ability to use triSYCL as the SYCL implementation in TensorFlow when OpenCL SYCL support is enabled, relates to #22 

Note that triSYCL is still work in progress and many features are missing, currently only the host device can be used (triSYCL/triSYCL#51).

ComputeCPP is a much more reliable way of using SYCL with TensorFlow.

### CODE REVIEW

**Review by benoitsteiner — APPROVED** [review]

**Comment by benoitsteiner** [comment]
Can you reformat tensorflow/core/platform/default/build_config/BUILD to fix the errors below ?
FAIL: buildifier found errors and/or warnings in above BUILD files.
buildifier suggested the following changes:
181c181
<         "-Wl,-rpath,../local_config_sycl/sycl/lib"
---
>         "-Wl,-rpath,../local_config_sycl/sycl/lib",
183,184c183,186
<     deps = if_ccpp(["@local_config_sycl//sycl:syclrt"],
<                    ["@local_config_sycl//sycl:sycl_headers"]),
---
>     deps = if_ccpp(
>         ["@local_config_sycl//sycl:syclrt"],
>         ["@local_config_sycl//sycl:sycl_headers"],
>     ),
209c211
< )
\ No newline at end of file
---
> )

**Comment by benoitsteiner** [comment]
Jenkins, test this please.

**Comment by benoitsteiner** [comment]
Buildifier is complaining about the following remaining issue:

tensorflow/core/platform/default/build_config/BUILD # reformat 
buildifier suggested the following changes:
183,184c183,185
<     deps = if_ccpp(["@local_config_sycl//sycl:syclrt"],
<                    ["@local_config_sycl//sycl:sycl_headers"]
---
>     deps = if_ccpp(
>         ["@local_config_sycl//sycl:syclrt"],
>         ["@local_config_sycl//sycl:sycl_headers"],
Please fix manually or run buildifier <file> to auto-fix.

**Comment by sb2nov** [comment]
@a-doumoulakis can you fix the merge conflict and take a look at the tests

**Comment by benoitsteiner** [comment]
Jenkins, test this please.

**Comment by benoitsteiner** [comment]
Unfortunately, we have another sanity check failure:
FAIL: buildifier found errors and/or warnings in above BUILD files.
buildifier suggested the following changes:
204,205c204,206
<     deps = if_ccpp(["@local_config_sycl//sycl:syclrt"],
<                    ["@local_config_sycl//sycl:sycl_headers"],
---
>     deps = if_ccpp(
>         ["@local_config_sycl//sycl:syclrt"],
>         ["@local_config_sycl//sycl:sycl_headers"],
Please fix manually or run buildifier <file> to auto-fix.

Maybe you could run [buildifier](https://github.com/bazelbuild/buildtools) on all the BUILD file to knock all these failures in one go ?

**Comment by benoitsteiner** [comment]
Jenkins, test this please.

**Comment by benoitsteiner** [comment]
I signed it

**Comment by gunan** [comment]
Jenkins, test this please.

**Comment by martinwicke** [comment]
Jenkins, test this please.

File: third_party/sycl/sycl/BUILD.tpl [hunk 1]
```diff
@@ -10,16 +10,27 @@ package(default_visibility = ["//visibility:public"])
...
         "define": "using_sycl=true",
+        "define": "using_trisycl=false",
     },
 )
 
+config_setting(
+    name = "using_sycl_trisycl",
+    values = {
+        "define": "using_sycl=true",
+        "define": "using_trisycl=true",
```
Comment (gunan) [inline third_party/sycl/sycl/BUILD.tpl]: Looks like bazel is not happy with this here.
I will roll this change back and then we can look into modifying this config setting.

**Pull request merged**